catalog/transformer/catalog-transformer-streaming-lib/src/test/java/org/codice/ddf/transformer/xml/streaming/lib/TestSaxEventToXmlElementConverter.java

```diff
@@ -0,0 +1,210 @@
+                + "            </bbb:baz>"
+                + "        </aaa:bar>"
+                + "    </aaa:foo>"
+                + "</outer>";
+                //@formatter:on
+        String reconstructedExpectation =
+                //@formatter:off
+                  "<outer xmlns='http://www.w3.com'>"
+                + "    <aaa:foo xmlns:aaa='whocares' name='outside'>"
+                + "        <aaa:bar>"
+                + "            <bbb:baz xmlns:bbb='inside1'>"
+                + "                <aaa:verybad xmlns:aaa='inside2' name='scope matters'></aaa:verybad>"
+                + "            </bbb:baz>"
+                + "        </aaa:bar>"
+                + "    </aaa:foo>"
+                + "</outer>";
+                //@formatter:on
+
+        TestSaxParser parser = new TestSaxParser();
+        String reconstructedXml = parser.parseAndReconstruct(snippet);
+        reconstructedExpectation = reconstructedExpectation.replaceAll("'", "\"");
+        assertThat(reconstructedXml, is(reconstructedExpectation));
+    }
+
+    @Test
+    public void testSaxEventToXmlConverterNormal()
+            throws XMLStreamException, IOException, SAXException {
+        String normalSnippet =
+                //@formatter:off
+                  "<?xml version='1.0'?>"
+                + "<!-- both namespace prefixes are available throughout -->"
+                + "<bk:book xmlns:bk='urn:loc.gov:books' xmlns:isbn='urn:ISBN:0-395-36341-6'>"
+                + "    <bk:title>Cheaper by the Dozen</bk:title>"
+                + "    <isbn:number>1568491379</isbn:number>"
+                + "</bk:book>";
+                //@formatter:on
+
+        String reconstructedExpectation =
+                //@formatter:off
+                  "<bk:book xmlns:bk='urn:loc.gov:books'>"
+                + "    <bk:title>Cheaper by the Dozen</bk:title>"
+                + "    <isbn:number xmlns:isbn='urn:ISBN:0-395-36341-6'>1568491379</isbn:number>"
+                + "</bk:book>";
+                //@formatter:on
+
+        TestSaxParser parser = new TestSaxParser();
+        String reconstructedXml = parser.parseAndReconstruct(normalSnippet);
+        reconstructedExpectation = reconstructedExpectation.replaceAll("'", "\"");
+        assertThat(reconstructedExpectation, is(reconstructedXml));
+    }
+
+    @Test
+    public void testSaxEventToXmlConverterScopedPrefixRedeclaration()
+            throws XMLStreamException, IOException, SAXException {
+        String snippet =
+                //@formatter:off
+                  "<?xml version='1.0'?>"
+                + "<!-- initially, the default namespace is 'books' -->"
+                + "<book xmlns='urn:loc.gov:books' xmlns:isbn='urn:ISBN:0-395-36341-6'>"
+                + "    <title>Cheaper by the Dozen</title>"
+                + "    <isbn:number>1568491379</isbn:number>"
+                + "    <notes><!-- make HTML the default namespace for some commentary -->"
```

Comment:
If you put it in the notes tag, it would apply to the notes tag as well, which is not what the original xml does. I think for the purposes of this test as well, it doesn't matter either way. We want to make sure that the converter does actually redeclare the namespace (so maybe the more the better ).